src/EnergyPlus/OutputProcessor.cc

```diff
@@ -6509,16 +6518,9 @@ void UpdateMeterReporting(EnergyPlusData &state)
                                                                  state.dataIPShortCut->cAlphaFieldNames,
                                                                  state.dataIPShortCut->cNumericFieldNames);
 
-        varnameLen = index(Alphas(1), '[');
-        if (varnameLen != std::string::npos) Alphas(1).erase(varnameLen);
-
-        ReportFreq = determineFrequency(state, Alphas(2));
-
-        int meterIndex = findMeterIndexFromMeterName(Alphas(1));
-        if (meterIndex > 0) {
-            // MeterFileOnlyIndicator is false, CumulativeIndicator is false
-            SetInitialMeterReportingAndOutputNames(state, meterIndex, false, ReportFreq, false);
-        } else {
+        bool meterFileOnlyIndicator = false;
+        bool cumulativeIndicator = false;
+        if (!setupMeterFromMeterName(Alphas(1), Alphas(2), meterFileOnlyIndicator, cumulativeIndicator)) {
```

Comment:
The usages are cleaner too. I could have made the ShowWarningError inside the lambda itself, but I find it clearer like this, and more versatile should we need to customize the warning message at some point